lib/websocket.js

```diff
@@ -188,8 +190,19 @@ function update() {
               treatment.x = timestamp.getTime();
               return treatment;
           });
-          // all done, do loadData
-          loadData( );
+
+          profiles.list(function (err, results) {
+              // There should be only one document in the profile collection with a DIA.  If there are multiple, use the last one.
+              results.forEach(function(element, index, array) {
+                  if (element) {
+                    if (element.dia) {
+                        profileData[0] = element;
+                    }
+                  }
+              });
```

Comment:
Based on the comment, it looks like you're accessing the last element of results using an O(n) approach. If so, consider using an O(1) implementation, such as:

```
if (results.length > 0) {
  var element = results[results.length - 1];
  if (element && element.dia) {
    profileData[0] = element;
  }
}
```
(and you may not even need the outermost `if`)